### DISCUSSION TITLE
How to correctly add language default lemmatizer into a blank model

### DISCUSSION THREAD

**Discussion by Pandalei97** [explosion/spaCy#8341] [Help: Coding & Implementations] [answered]
Hi, I'm having problem by adding language default lemmatizer in to a blank model.

My blank model is initialized by `python -m spacy init vectors fr {vector_path} {blank_model_path}`

My model (simplified for the question):
```python
nlp = spacy.load({blank_model_path})
nlp.add_pipe("lemmatizer").initialize()

doc = nlp("Je suis étudiante") # "Je suis étudiante" == "I'm a student"
lemma = [t.lemma_ for t in doc]
```

I expect to have `['je', 'être', 'étudiant']`
But I got `['je', 'suis', 'étudiante']`

The lemmatizer just takes the lower form of each word.

Did I do something wrong ? Thanks in advance for your reply.

[Environment information]
Spacy version : 3.0.6
Python version: 3.6.5

**Comment by adrianeboyd** [comment, marked as answer]
Many of the default lemmatizers require `token.pos` from a previous component, so you usually need either `tagger`+`attribute_ruler` or `morphologizer`. The default rule-based lemmatizer shows a warning if `token.pos` missing, but some of the language-specific ones like the French lemmatizer might not have the warning at this point.

In a blank pipeline like this, one option is a lookup lemmatizer, which you can add as shown in this section under "Switch from rule-based to lookup lemmatization":

https://spacy.io/models#design-modify

**Reply by Pandalei97** [reply]
Thanks for your response. I thought that lookup was set as the default value as mentioned in your api page: https://spacy.io/api/lemmatizer#config-and-implementation. 
> The lemmatizer mode, e.g. "lookup" or "rule". Defaults to "lookup".

Maybe the defeult value is changed somewhere in the French lemmatizer.

**Reply by adrianeboyd** [reply]
That's a good point about the documentation, which is definitely incorrect/confusing here.

Languages that don't have any custom lemmatizer settings will have have `lookup` as a default, but many language have custom settings or a custom lemmatizer with its own defaults. If we have anything better than a lookup table (rule-based tables, pos-based lookup lists, a third party lemmatizer), the default is usually not `lookup`, but you don't see any of these internal details when you call `nlp.add_pipe("lemmatizer")`.

You can always check `nlp.get_pipe("lemmatizer").mode` to see what the mode is, but you're right that the defaults are really unclear. We should probably add a table with the language-specific defaults to the docs.
